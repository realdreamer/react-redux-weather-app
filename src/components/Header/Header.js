import React from "react";
import SearchForm from "../SearchForm/SearchForm";

const Header = props => (
  <header>
    <h1>Weather App</h1>
    <SearchForm onSearch={props.onSearch} />
  </header>
);

export default Header;
