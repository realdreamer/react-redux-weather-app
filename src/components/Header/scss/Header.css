.header {
  background-color: #f7b18e;
  color: #fff;
  padding: 10px;
  text-align: center; }

.header-title {
  font-size: 18px;
  text-transform: uppercase; }
