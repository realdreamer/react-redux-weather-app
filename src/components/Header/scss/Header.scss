$header-bg-color: #f7b18e;
.header {
  background-color: $header-bg-color;
  color: #fff;
  padding: 10px;
  text-align: center;
}
.header-title {
  font-size: 18px;
  text-transform: uppercase;
}
