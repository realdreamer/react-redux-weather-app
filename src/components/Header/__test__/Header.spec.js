import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Header from '../Header';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer
                  .create(<Header />)
                  .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should attach props pass in', () => {
    const tree = renderer
                  .create(<Header />)
                  .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
