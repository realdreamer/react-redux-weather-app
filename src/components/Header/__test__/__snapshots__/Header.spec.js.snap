// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <h1>
    Weather App
  </h1>
  <div
    className="header-search"
  >
    <form
      className="city-form"
      id="city-form"
      name="city-form"
      onSubmit={[Function]}
    >
      <input
        id="city-search"
        name="city-search__input"
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        id="city-search-btn"
        name="city-search__btn"
        type="submit"
        value="Search"
      />
    </form>
  </div>
</header>
`;

exports[`Header should attach props pass in 1`] = `
<header>
  <h1>
    Weather App
  </h1>
  <div
    className="header-search"
  >
    <form
      className="city-form"
      id="city-form"
      name="city-form"
      onSubmit={[Function]}
    >
      <input
        id="city-search"
        name="city-search__input"
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        id="city-search-btn"
        name="city-search__btn"
        type="submit"
        value="Search"
      />
    </form>
  </div>
</header>
`;
